src/Workspaces/SharedUtilitiesAndExtensions/Compiler/VisualBasic/Services/SyntaxFacts/VisualBasicSyntaxFacts.vb

```diff
@@ -2173,6 +2173,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.LanguageServices
                     Return DeclarationKind.NamespaceImport
                 Case SyntaxKind.ClassBlock
                     Return DeclarationKind.Class
+                Case SyntaxKind.ClassStatement
+                    If Not IsChildOf(declaration, SyntaxKind.ClassBlock) Then
```

Comment:
Why is this condition needed?

Reply:
Without it break few tests. I followed the same logic I found in the code bellow, where have XXBlock and XXStatement.

Reply:
1. which tests break?
2. we're now very inconsistent in this helper.  for example, how we handle structs and interfaces.
3. it's not clear ot me why the fix is here, and not in the feature using this helper.

Reply:
> 1. which tests break?

```
✘ Microsoft.CodeAnalysis.VisualBasic.UnitTests.Editting.SyntaxGeneratorTests.TestMultiFieldMembers
Error: Assert.Equal() Failure
Expected: 3
Actual:   0
Stack trace:
   em Xunit.Assert.Equal[T](T expected, T actual, IEqualityComparer`1 comparer) na C:\projects\xunit\src\xunit.assert\Asserts\EqualityAsserts.cs:linha 40
   em Xunit.Assert.Equal[T](T expected, T actual) na C:\projects\xunit\src\xunit.assert\Asserts\EqualityAsserts.cs:linha 24
   em Microsoft.CodeAnalysis.VisualBasic.UnitTests.Editting.SyntaxGeneratorTests.TestMultiFieldMembers() na D:\Devel\git\roslyn\src\Workspaces\VisualBasicTest\CodeGeneration\SyntaxGeneratorTests.vb:linha 3323


✘ Microsoft.CodeAnalysis.VisualBasic.UnitTests.Editting.SyntaxGeneratorTests.TestMultiAttributes
Error: Assert.IsAssignableFrom() Failure
Expected: typeof(Microsoft.CodeAnalysis.VisualBasic.Syntax.ClassBlockSyntax)
Actual:   typeof(Microsoft.CodeAnalysis.VisualBasic.Syntax.ClassStatementSyntax)
Stack trace:
   em Xunit.Assert.IsAssignableFrom(Type expectedType, Object object) na C:\projects\xunit\src\xunit.assert\Asserts\TypeAsserts.cs:linha 38
   em Microsoft.CodeAnalysis.VisualBasic.UnitTests.Editting.SyntaxGeneratorTests.VerifySyntax[TSyntax](SyntaxNode type, String expectedText) na D:\Devel\git\roslyn\src\Workspaces\VisualBasicTest\CodeGeneration\SyntaxGeneratorTests.vb:linha 41
   em Microsoft.CodeAnalysis.VisualBasic.UnitTests.Editting.SyntaxGeneratorTests.TestMultiAttributes() na D:\Devel\git\roslyn\src\Workspaces\VisualBasicTest\CodeGeneration\SyntaxGeneratorTests.vb:linha 3539
```

> 2. we're now very inconsistent in this helper. for example, how we handle structs and interfaces.

I agree with you. I tried to not do that, but could not find better fix.

> 3. it's not clear ot me why the fix is here, and not in the feature using this helper.

To be honest, I don't know if is the correct place or no, this is my first try on contributing to this repo, I'm trying to learn as much as I can. Is very challenging.